server/models/Collective.js

```diff
@@ -1835,30 +1835,42 @@ export default function (Sequelize, DataTypes) {
     const isHost = await this.isHost();
     if (isHost) {
       // We only expect currency change at the beginning of the history of the Host
-      const transactionCount = await models.Transaction.count({ where: { HostCollectiveId: this.id } });
+      // We're however good with it if currency is already recorded as hostCurrency in the ledger
+      const transactionCount = await models.Transaction.count({
+        where: { HostCollectiveId: this.id, hostCurrency: { [Op.not]: currency } },
+      });
       if (transactionCount > 0) {
         throw new Error(
           'You cannot change the currency of an Host with transactions. Please contact support@opencollective.com.',
         );
       }
-      const collectives = await this.getHostedCollectives();
+      let collectives = await this.getHostedCollectives();
+      collectives = collectives.filter(collective => collective.id !== this.id);
       // We use setCurrency so that it will cascade to Tiers
-      await Promise.map(collectives, collective => collective.setCurrency(currency), { concurrency: 3 });
+      if (collectives.length > 0) {
+        await Promise.map(collectives, collective => collective.setCurrency(currency), { concurrency: 3 });
+      }
+    }
+
+    // Update all transactions
+    const transactions = await models.Transaction.findAll({ where: { CollectiveId: this.id } });
+    if (transactions.length > 0) {
+      await Promise.map(transactions, transaction => transaction.setCurrency(currency), { concurrency: 3 });
```

Comment:
This query will only return one transaction of the pair, we should make sure to update both, either here or in `transaction.setCurrency()`.

Reply:
No, the goal is to have a consistent currency regarding the CollectiveId. We should not necessarily update the other one because it has a different CollectiveId and maybe a different desired currency.

Reply:
I don't think allowing a transaction pair to have a different currency is a good idea. Shouldn't the Credit/Debit pair always be settled in the same currency?

Reply:
Within the current design, I think we have to, because we don't want to allow the "oppositeTransaction" to be in a different currency than the collective profile. The risk is to make balance computation impossible. See current versions of `getBalances` which is not handling multi-currency.

Happy to discuss alternatives, this is the right moment.